framework/include/utils/RankTwoTensor.h

```diff
@@ -170,6 +170,9 @@ class RankTwoTensor
   RankTwoTensor operator - () const;
 
   /// performs _vals *= a
```

Comment:
Please fix the comment to actually describe the operator. I'm not convinced that we should implement this overload at all. It is not obvious what it should do. Should it maybe rather set the tensor to I*a (I being the identity matrix)? 

Are you using this for anything other than `tensor = 0.0`? If not then please use `tensor.zero()` and remove this method.